4.x: Building the Native Docker image fails

[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 4.0.0
* Helidon MP
* JDK version: 21
* OS: macOS 13
* Docker version (if applicable):  24.0.6

----------

## Problem Description
[//]: # "`docker build -t helidon-quickstart-mp-native -f Dockerfile.native .` fails"
[//]: # "State if the problem is easily reproducible. It fails every time I run the command above"
[//]: # "Include stack traces or command outputs"
```
19.67 [ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.27:compile (build-native-image) on project helidon-quickstart-mp: Execution of /opt/graalvm-community-java21/bin/native-image @target/tmp/native-image-1030896273986161271.args io.helidon.Main returned non-zero result -> [Help 1]
19.67 [ERROR]
19.67 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
19.67 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
19.67 [ERROR]
19.67 [ERROR] For more information about the errors and possible solutions, please read the following articles:
19.67 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
------
Dockerfile.native:21
--------------------
  19 |     # the pom
  20 |     ADD pom.xml .
  21 | >>> RUN mvn package -Pnative-image -Dnative.image.skip -Dmaven.test.skip -Declipselink.weave.skip
  22 |
  23 |     # Do the Maven build!
--------------------
ERROR: failed to solve: process "/bin/sh -c mvn package -Pnative-image -Dnative.image.skip -Dmaven.test.skip -Declipselink.weave.skip" did not complete successfully: exit code: 1
```

## Steps to reproduce
[//]: # "Step by step instructions to reproduce the problem"
[//]: # "Provide sample code/application if relevant"
1. Generate the project using the Helidon MP Quickstart Maven archetype 
```
mvn -U archetype:generate -DinteractiveMode=false \
    -DarchetypeGroupId=io.helidon.archetypes \
    -DarchetypeArtifactId=helidon-quickstart-mp \
    -DarchetypeVersion=4.0.0 \
    -DgroupId=io.helidon.examples \
    -DartifactId=helidon-quickstart-mp \
    -Dpackage=io.helidon.examples.quickstart.mp
```
2. 
```
cd helidon-quickstart-mp
mvn package
```
3. ```docker build -t helidon-quickstart-mp-native -f Dockerfile.native .```

--- a/archetypes/helidon/src/main/archetype/common/docker.xml
+++ b/archetypes/helidon/src/main/archetype/common/docker.xml
@@ -62,11 +62,11 @@
                     </includes>
                 </templates>
                 <model>
-                    <list key="poms">
-                        <value>pom.xml</value>
+                    <list key="poms" if="!${multi-module}">
+                        <value>ADD pom.xml .</value>
                     </list>
                     <list key="src-dirs" if="!${multi-module}">
-                        <value>src</value>
+                        <value>ADD src src</value>
                     </list>
                     <list key="native-sections" if="${docker.native-image}">
                         <value file="files/README.native.md"/>

--- a/archetypes/helidon/src/main/archetype/common/files/Dockerfile.jlink.mustache
+++ b/archetypes/helidon/src/main/archetype/common/files/Dockerfile.jlink.mustache
@@ -1,6 +1,16 @@
 
 # 1st stage, build the app
-FROM maven:3.8.4-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+WORKDIR /usr/share
+
+# Install maven
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 

--- a/archetypes/helidon/src/main/archetype/common/files/Dockerfile.mustache
+++ b/archetypes/helidon/src/main/archetype/common/files/Dockerfile.mustache
@@ -17,14 +17,14 @@ WORKDIR /helidon
 # Incremental docker builds will always resume after that, unless you update
 # the pom
 {{#poms}}
-ADD {{.}} {{.}}
+{{.}}
 {{/poms}}
 RUN mvn package -Dmaven.test.skip -Declipselink.weave.skip {{#docker-phase1-options}}{{.}}{{^last}} {{/last}}{{/docker-phase1-options}}
 
 # Do the Maven build!
 # Incremental docker builds will resume here when you change sources
 {{#src-dirs}}
-ADD {{.}} {{.}}
+{{.}}
 {{/src-dirs}}
 RUN mvn package -DskipTests
 

--- a/archetypes/helidon/src/main/archetype/common/files/pom.xml.mustache
+++ b/archetypes/helidon/src/main/archetype/common/files/pom.xml.mustache
@@ -15,11 +15,11 @@
     <artifactId>{{artifactId}}</artifactId>
     <version>{{project-version}}</version>
 
-    <properties>
 {{#mainClass}}
+    <properties>
         <mainClass>{{mainClass}}</mainClass>
-{{/mainClass}}
     </properties>
+{{/mainClass}}
 
     <dependencies>
         {{#dependencies}}

--- a/archetypes/helidon/src/main/archetype/mp/oci/oci-mp.xml
+++ b/archetypes/helidon/src/main/archetype/mp/oci/oci-mp.xml
@@ -397,13 +397,14 @@ Mount this volume as part of your application containers specification
                                 ]]></value>
             </list>
             <list key="poms">
-                <value>client/pom.xml</value>
-                <value>server/pom.xml</value>
+                <value>ADD pom.xml pom.xml</value>
+                <value>ADD client/pom.xml client/pom.xml</value>
+                <value>ADD server/pom.xml server/pom.xml</value>
             </list>
             <list key="src-dirs">
-                <value>client/src</value>
-                <value>server/src</value>
-                <value>spec/api.yaml</value>
+                <value>ADD client/src client/src</value>
+                <value>ADD server/src server/src</value>
+                <value>ADD spec/api.yaml spec/api.yaml</value>
             </list>
         </model>
     </output>

--- a/docs/mp/guides/mp-tutorial.adoc
+++ b/docs/mp/guides/mp-tutorial.adoc
@@ -948,7 +948,17 @@ Add a new `Dockerfile` in the project root directory with the following content:
 [source,bash]
 .Dockerfile content
 ----
-FROM maven:3.8.4-openjdk-17-slim as build <1>
+FROM container-registry.oracle.com/java/openjdk:21 as build <1>
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
+
 WORKDIR /helidon
 
 ADD pom.xml .
@@ -958,13 +968,13 @@ ADD src src
 RUN mvn package -DskipTests <3>
 RUN echo "done!"
 
-FROM openjdk:17-jdk-slim <4>
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
-COPY --from=build /helidon/target/helidon-mp-tutorial.jar ./ <5>
+COPY --from=build /helidon/target/helidon-mp-tutorial.jar ./ <4>
 COPY --from=build /helidon/target/libs ./libs
 
-CMD ["java", "-jar", "helidon-mp-tutorial.jar"] <6>
+CMD ["java", "-jar", "helidon-mp-tutorial.jar"] <5>
 EXPOSE 8080
 ----
 
@@ -977,9 +987,8 @@ EXPOSE 8080
  builds faster because they will use this cached layer rather than downloading
  everything again.
 <3> Add the source code and do the real build.
-<4> Start a second stage using a much smaller runtime image.
-<5> Copy the binary and libraries from the first stage.
-<6> Set the initial command and expose port 8080.
+<4> Copy the binary and libraries from the first stage.
+<5> Set the initial command and expose port 8080.
 
 To create the Docker image, use the following command:
 

--- a/examples/employee-app/Dockerfile
+++ b/examples/employee-app/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2019, 2021 Oracle and/or its affiliates.
+# Copyright (c) 2019, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.6.3-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -33,7 +42,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/integrations/cdi/datasource-hikaricp/Dockerfile
+++ b/examples/integrations/cdi/datasource-hikaricp/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2021 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.6.3-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -32,7 +41,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/quickstarts/helidon-quickstart-mp/Dockerfile
+++ b/examples/quickstarts/helidon-quickstart-mp/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2022 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.8.4-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -32,7 +41,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/quickstarts/helidon-quickstart-mp/Dockerfile.jlink
+++ b/examples/quickstarts/helidon-quickstart-mp/Dockerfile.jlink
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020, 2022 Oracle and/or its affiliates.
+# Copyright (c) 2020, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.8.4-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 

--- a/examples/quickstarts/helidon-quickstart-se/Dockerfile.jlink
+++ b/examples/quickstarts/helidon-quickstart-se/Dockerfile.jlink
@@ -15,7 +15,7 @@
 #
 
 # 1st stage, build the app
-FROM container-registry.oracle.com/java/openjdk:21 as maven
+FROM container-registry.oracle.com/java/openjdk:21 as build
 
 # Install maven
 WORKDIR /usr/share

--- a/examples/quickstarts/helidon-standalone-quickstart-mp/Dockerfile
+++ b/examples/quickstarts/helidon-standalone-quickstart-mp/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2022 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.8.4-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -32,7 +41,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/quickstarts/helidon-standalone-quickstart-mp/Dockerfile.jlink
+++ b/examples/quickstarts/helidon-standalone-quickstart-mp/Dockerfile.jlink
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020, 2022 Oracle and/or its affiliates.
+# Copyright (c) 2020, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.8.4-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -35,6 +44,6 @@ RUN echo "done!"
 
 FROM debian:stretch-slim
 WORKDIR /helidon
-COPY --from=build /helidon/target/helidon-quickstart-mp-jri ./
+COPY --from=build /helidon/target/helidon-standalone-quickstart-mp-jri ./
 ENTRYPOINT ["/bin/bash", "/helidon/bin/start"]
 EXPOSE 8080

--- a/examples/quickstarts/helidon-standalone-quickstart-mp/README.md
+++ b/examples/quickstarts/helidon-standalone-quickstart-mp/README.md
@@ -106,13 +106,13 @@ Start the application:
 Build the "native" Docker Image
 
 ```
-docker build -t helidon-quickstart-mp-native -f Dockerfile.native .
+docker build -t helidon-standalone-quickstart-mp-native -f Dockerfile.native .
 ```
 
 Start the application:
 
 ```
-docker run --rm -p 8080:8080 helidon-quickstart-mp-native:latest
+docker run --rm -p 8080:8080 helidon-standalone-quickstart-mp-native:latest
 ```
   
 ## Build a Java Runtime Image using jlink
@@ -141,7 +141,7 @@ See https://github.com/oracle/helidon-build-tools/tree/master/helidon-maven-plug
 Start the application:
 
 ```
-./target/helidon-quickstart-se-jri/bin/start
+./target/helidon-standalone-quickstart-mp-jri/bin/start
 ```
 
 ### Multi-stage Docker build
@@ -149,17 +149,17 @@ Start the application:
 Build the JRI as a Docker Image
 
 ```
-docker build -t helidon-quickstart-mp-jri -f Dockerfile.jlink .
+docker build -t helidon-standalone-quickstart-mp-jri -f Dockerfile.jlink .
 ```
 
 Start the application:
 
 ```
-docker run --rm -p 8080:8080 helidon-quickstart-mp-jri:latest
+docker run --rm -p 8080:8080 helidon-standalone-quickstart-mp-jri:latest
 ```
 
 See the start script help:
 
 ```
-docker run --rm helidon-quickstart-mp-jri:latest --help
+docker run --rm helidon-standalone-quickstart-mp-jri:latest --help
 ```

--- a/examples/quickstarts/helidon-standalone-quickstart-se/Dockerfile
+++ b/examples/quickstarts/helidon-standalone-quickstart-se/Dockerfile
@@ -15,7 +15,7 @@
 #
 
 # 1st stage, build the app
-FROM container-registry.oracle.com/java/openjdk:21 as maven
+FROM container-registry.oracle.com/java/openjdk:21 as build
 
 # Install maven
 WORKDIR /usr/share

--- a/examples/quickstarts/helidon-standalone-quickstart-se/Dockerfile.jlink
+++ b/examples/quickstarts/helidon-standalone-quickstart-se/Dockerfile.jlink
@@ -15,7 +15,7 @@
 #
 
 # 1st stage, build the app
-FROM container-registry.oracle.com/java/openjdk:21 as maven
+FROM container-registry.oracle.com/java/openjdk:21 as build
 
 # Install maven
 WORKDIR /usr/share
@@ -44,6 +44,6 @@ RUN echo "done!"
 
 FROM debian:stretch-slim
 WORKDIR /helidon
-COPY --from=build /helidon/target/helidon-quickstart-se-jri ./
+COPY --from=build /helidon/target/helidon-standalone-quickstart-se-jri ./
 ENTRYPOINT ["/bin/bash", "/helidon/bin/start"]
 EXPOSE 8080

--- a/examples/quickstarts/helidon-standalone-quickstart-se/README.md
+++ b/examples/quickstarts/helidon-standalone-quickstart-se/README.md
@@ -142,7 +142,7 @@ See https://github.com/oracle/helidon-build-tools/tree/master/helidon-maven-plug
 Start the application:
 
 ```
-./target/helidon-quickstart-se-jri/bin/start
+./target/helidon-standalone-quickstart-se-jri/bin/start
 ```
 
 ### Multi-stage Docker build
@@ -150,17 +150,17 @@ Start the application:
 Build the JRI as a Docker Image
 
 ```
-docker build -t helidon-quickstart-se-jri -f Dockerfile.jlink .
+docker build -t helidon-standalone-quickstart-se-jri -f Dockerfile.jlink .
 ```
 
 Start the application:
 
 ```
-docker run --rm -p 8080:8080 helidon-quickstart-se-jri:latest
+docker run --rm -p 8080:8080 helidon-standalone-quickstart-se-jri:latest
 ```
 
 See the start script help:
 
 ```
-docker run --rm helidon-quickstart-se-jri:latest --help
+docker run --rm helidon-standalone-quickstart-se-jri:latest --help
 ```

--- a/examples/translator-app/README.md
+++ b/examples/translator-app/README.md
@@ -28,7 +28,7 @@ kubectl apply \
 With Docker:
 ```bash
 docker build -t helidon-examples-translator-backend backend/
-docker build -t helidon-examples--translator-frontend frontend/
+docker build -t helidon-examples-translator-frontend frontend/
 docker run --rm -d -p 9080:9080 \
     --link zipkin \
     --name helidon-examples-translator-backend \

--- a/examples/translator-app/backend/Dockerfile
+++ b/examples/translator-app/backend/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2021 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.6.3-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -33,7 +42,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/translator-app/backend/pom.xml
+++ b/examples/translator-app/backend/pom.xml
@@ -29,7 +29,6 @@
     </parent>
     <groupId>io.helidon.examples.translator</groupId>
     <artifactId>helidon-examples-translator-backend</artifactId>
-    <version>4.0.0-SNAPSHOT</version>
     <name>Helidon Examples Translator Backend</name>
 
     <description>

--- a/examples/translator-app/frontend/Dockerfile
+++ b/examples/translator-app/frontend/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2021 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.6.3-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -33,7 +42,7 @@ RUN mvn package -Dmaven.test.skip
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

--- a/examples/translator-app/frontend/pom.xml
+++ b/examples/translator-app/frontend/pom.xml
@@ -29,7 +29,6 @@
     </parent>
     <groupId>io.helidon.examples.translator</groupId>
     <artifactId>helidon-examples-translator-frontend</artifactId>
-    <version>4.0.0-SNAPSHOT</version>
     <name>Helidon Examples Translator Frontend</name>
 
     <description>

--- a/examples/webserver/opentracing/Dockerfile
+++ b/examples/webserver/opentracing/Dockerfile
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018, 2021 Oracle and/or its affiliates.
+# Copyright (c) 2018, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -15,7 +15,16 @@
 #
 
 # 1st stage, build the app
-FROM maven:3.6.3-openjdk-17-slim as build
+FROM container-registry.oracle.com/java/openjdk:21 as build
+
+# Install maven
+WORKDIR /usr/share
+RUN set -x && \
+    curl -O https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz && \
+    tar -xvf apache-maven-*-bin.tar.gz  && \
+    rm apache-maven-*-bin.tar.gz && \
+    mv apache-maven-* maven && \
+    ln -s /usr/share/maven/bin/mvn /bin/
 
 WORKDIR /helidon
 
@@ -33,7 +42,7 @@ RUN mvn package -DskipTests
 RUN echo "done!"
 
 # 2nd stage, build the runtime image
-FROM openjdk:17-jdk-slim
+FROM container-registry.oracle.com/java/openjdk:21
 WORKDIR /helidon
 
 # Copy the binary built in the 1st stage

